versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -588,29 +593,29 @@ protected Hash mergeTransplantCommon(
           result);
     }
 
-    if (params.isDryRun()) {
-      return toHead;
-    }
-
     if (params.keepIndividualCommits()) {
       // re-apply commits in 'sequenceToTransplant' onto 'targetBranch'
-      toHead =
+      Hash newHead =
           copyCommits(
               ctx,
               timeInMicros,
               toHead,
               commitsToMergeChronological,
               newKeyLists,
               params.getUpdateCommitMetadata(),
-              mergePredicate);
-
-      // Write commits
+              mergePredicate,
+              addedCommits);
+
+      if (!params.isDryRun()) {
```

Comment:
Would it break anything if we leave this `if` where it was before?
I don't mind having the copyCommits doing that in the old model (because it eventually goes away), if it's a lot of work.